### ISSUE TITLE
Can we not just have some unjudged simple responses.

### ISSUE THREAD

**Issue by SomeProgrammerGuy** [Alamofire/Alamofire#3530]
## Problem

I have been programming for a long time but I’m new (ish) to Swift and SwiftUI for note.

I just wanted to upload some files.

But honestly, ignoring that pretty much all the online examples (not just yours and yes I understand they are out of your control)  are mostly out of date or just wrong and simply don’t work (which happens when you change stuff). I get this but honestly please fix you examples (or even maybe include some). Not just weird documentation that explains each part without context. Lots of actual real examples doing it.

You really didn’t go for the KISS principle did you.

I have never seen anything that is so over engineered and out there as to getting the actual response. I’m sure it’s probably in there somewhere but after days with fighting with your library I’m (mostly) not sure where.

The truth is REST APIs’ are supposed to be a standard but in real life they really are not. They are interpreted with a lot of different results.

Over many languages I have learned in mind of REST API’s there is two pieces of data that folk require to be returned from REST API’s.

    a. The HTTP(s) Response code .
    b. Did it return some text (in some format I will have to work out for myself, YOU cannot predict this as it is so subjective RFC’s or 
        not, in fact even some 204 responses *wrongly* returning data sort of proving my bigger picture point)?

If you want to keep all that over convoluted stuff then hey why not have an additional .basicResponse that just contains the HTTP(s) response and what was returned (or not) in whatever format they returned some data in (or not). I imagine that this will end up what most folk actually end up using rather than all this over engineered stuff. (Maybe you have one but honestly, I haven’t found it explained well (or even found it)).

I really don’t understand why you are including all this overengineered *stuff. That weird .validate() just seems like the most bizarre preachy thing ever. In most real-life API’s (that are wrong a lot, but as developers we have no choice) it just doesn’t work.

## Feature Request

It should be:

What was the HTTP(s) response code (if any, none would be unusual)?

Do something on that Response. (And yes, it will be up to the user to deal with that, not you). But do provide lots of examples.

From that response (I will have to convert myself to *whatever*). Most programmers are pretty much capable of this (debatably lol).

Noting this shouldn't be hidden in all the over engineered stuff. This should just be a simple .basicResponse that is obvious and really simple to use.

Maybe roughly :

.basicResponse.StatusCode

.basicResponse.Headers

.basicResponse.RawResponse

Etc. etc. you get the idea.

That’s it. What’s so complicated about that.

## Value to Alamofire

I think it speaks for itself.

**Comment by jshier** [comment]
It's good manners to be polite when asking for help.

As for examples, we have several simple examples in our documentation. Have you reviewed our [upload documentation](https://github.com/Alamofire/Alamofire/blob/master/Documentation/Usage.md#uploading-data-to-a-server)? It has examples of each method of uploading files, including multipart form uploads.

Did you actually try writing any code? The autocomplete should've given you a listing of all of our methods, and we have extensive inline and generated documentation, so you should be able to find whatever response handler you need. So do you have a specific question?

**Comment by SomeProgrammerGuy** [comment]
@jshier 

"It's good manners to be polite when asking for help."

I didn't actually ask you for any help. This was a "Feature Request".

I have also read through your "documentation" over many hours. Ignoring that it didn't answer any of my issues it doesn't even relate to the post I have given (unless there is something I have missed as to my comment).

And yes I wrote a lot of code over many hours that still I mostly had to work around as yours simply was to over complicated and engineered... in fact I will probably end up writing my own http(s) upload now which is a shame. Did you even read my request properly?

Again, I didn't ask a question. It was a "Feature Request".

The truth is if I had been involved in giving my time for *FREE* to some software (and your time is worthy) I would want folk to use my library.  Most folk don't give any feedback (they just do it another way) which is a shame as it's *YOUR* time that is wasted (and I think that's a real shame). 

The reason most folk don't give feedback is they mostly get ar*sy responses like this. 

Most folk just give up after 20 minutes of not understanding your documentation and do it another way. I'd rather your time wasn't wasted and you had a library that folk can mostly understand and use. 

You only have to do a simple search on StackOverflow to see how many folk really don't understand this library which is a shame as it seems to have had a lot of *FREE* work done to it.

I'm not trying to be argumentative here and I think free software is a great thing. But when a user says they can't understand it then they can't. It's not about how great you think it is. 

If you are one of the developers then that documentation will totally make sense to you. 

I have been in your shoes many times and over many years I have learned that if the *customer* so to speak says they don't get it then they are right and you are wrong.
